yb-voyager/src/queryparser/unsupportedConstructs.go

```diff
@@ -315,8 +328,115 @@ func containsSystemColumnsInWhereClause(whereClause *pg_query.Node) bool {
 	return false
 }
 
-// TODO: Implement
 func (qp *QueryParser) containsXmlFunctions() bool {
+	selectStmtNode, isSelectStmt := qp.ParseTree.Stmts[0].Stmt.Node.(*pg_query.Node_SelectStmt)
+	// Note: currently considering only SELECTs
+	if !isSelectStmt {
+		return false
+	}
+
+	if containsXmlFunctionsInTargetList(selectStmtNode.SelectStmt.TargetList) {
+		return true
+	}
+
+	if containsXmlFunctionsInFromClause(selectStmtNode.SelectStmt.FromClause) {
+		return true
+	}
+
+	if containsXmlFunctionsInWhereClause(selectStmtNode.SelectStmt.WhereClause) {
+		return true
+	}
+	return false
+}
+
+/*
+Query: SELECT id, xmlelement(name "employee", name) AS employee_data FROM employees
+ParseTree: version:160001  stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"id"}}  location:7}}  location:7}}
+target_list:{res_target:{name:"employee_data"  val:{xml_expr:{op:IS_XMLELEMENT  name:"employee"  args:{column_ref:{fields:{string:{sval:"name"}}  location:39}}  xmloption:XMLOPTION_DOCUMENT  location:11}}  location:11}}
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:67}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsXmlFunctionsInTargetList(targetList []*pg_query.Node) bool {
+	for _, target := range targetList {
+		resTarget := target.GetResTarget()
+		if resTarget == nil {
+			return false
+		}
+
+		xmlExpr := resTarget.Val.GetXmlExpr()
+		if xmlExpr != nil {
+			return true
+		}
+	}
+	return false
```

Comment:
we can also have xml functions in the target list e.g. `SELECT id, xpath('/person/name/text()', data) AS name from xml_example;` ?

Reply:
done